dot/state/block_finalisation.go

```diff
@@ -190,9 +193,17 @@ func (bs *BlockState) deleteFromTries(lastFinalised common.Hash) error {
 	if err != nil {
 		return fmt.Errorf("unable to retrieve header for last finalised block, hash: %s, err: %s", bs.lastFinalised, err)
 	}
-	stateRootTrie := bs.tries.get(lastFinalisedHeader.StateRoot)
+
+	stateRootTrie, err := bs.trieDB.Get(lastFinalisedHeader.StateRoot)
+	if err != nil {
+		return fmt.Errorf("unable to retrieve state root trie, hash: %s, err: %s", bs.lastFinalised, err)
+	}
+
 	if stateRootTrie != nil {
-		bs.tries.delete(lastFinalisedHeader.StateRoot)
+		err := bs.trieDB.Delete(lastFinalisedHeader.StateRoot)
```

Comment:
can just delete it directly without getting it from the database?

Reply:
Good catch, I don't know if there is a reason to check it before trying to delete it, I'm going to remove that check